components/overflow-group/overflow-group-mixin.js

```diff
@@ -0,0 +1,349 @@
+
+	static get properties() {
+		return {
+			/**
+			 * Use predefined classes on slot elements to set min and max buttons to show
+			 * @type {boolean}
+			 */
+			autoShow: {
+				type: Boolean,
+				attribute: 'auto-show',
+			},
+			/**
+			 * @ignore
+			 */
+			chompIndex: {
+				type: Number,
+			},
+			/**
+			 * minimum amount of buttons to show
+			 * @type {number}
+			 */
+			minToShow: {
+				type: Number,
+				reflect: true,
+				attribute: 'min-to-show',
+			},
+			/**
+			 * maximum amount of buttons to show
+			 * @type {number}
+			 */
+			maxToShow: {
+				type: Number,
+				reflect: true,
+				attribute: 'max-to-show',
+			},
+			/**
+			 * @ignore
+			 */
+			mini: {
+				type: Boolean,
+				reflect: true
+			},
+			/**
+			 * Set the opener type to 'icon' for a `...` menu icon instead of `More actions` text
+			 * @type {'default'|'icon'}
+			 */
+			openerType: {
+				type: String,
+				attribute: 'opener-type'
+			}
+		};
+	}
+
+	static get styles() {
+		return [offscreenStyles, css`
+			:host {
+				display: block;
+			}
+			:host([hidden]) {
+				display: none;
+			}
+			.d2l-overflow-group-container {
+				display: flex;
+				flex-wrap: wrap;
+				justify-content: var(--d2l-overflow-group-justify-content, normal);
+			}
+			.d2l-overflow-group-container ::slotted([data-is-chomped]) {
+				display: none !important;
+			}
+		`];
+	}
+
+	constructor() {
+		super();
+		this._handleItemMutation = this._handleItemMutation.bind(this);
+		this._handleResize = this._handleResize.bind(this);
+
+		this._throttledResize = (entries) => requestAnimationFrame(() => this._handleResize(entries));
+
+		this._overflowHidden = false;
+		this.autoShow = false;
+		this.maxToShow = -1;
+		this.minToShow = 1;
+		this.mini = this.openerType === OPENER_TYPE.ICON;
+		this.openerType = OPENER_TYPE.DEFAULT;
+		this._resizeObserver = null;
+		this._slotItems = [];
+	}
+
+	disconnectedCallback() {
+		super.disconnectedCallback();
+		if (this._resizeObserver) this._resizeObserver.disconnect();
+	}
+
+	async firstUpdated() {
+		super.firstUpdated();
+
+		// selected elements
+		this._buttonSlot = this.shadowRoot.querySelector('slot');
+
+		this._container = this.shadowRoot.querySelector('.d2l-overflow-group-container');
+
+		this._resizeObserver = new ResizeObserver(this._throttledResize);
+		this._resizeObserver.observe(this._container);
+	}
+
+	render() {
+		const overflowMenu = this.getOverflowMenu();
+
+		this._slotItems.forEach((element, index) => {
+			if (!this.overflowMenuHidden && index >= this.chompIndex) {
+				element.setAttribute('data-is-chomped', '');
+			} else {
+				element.removeAttribute('data-is-chomped');
+			}
+		});
+
+		return html`
+			<div class="d2l-overflow-group-container">
+				<slot @slotchange="${this._handleSlotChange}"></slot>
+				${overflowMenu}
+			</div>
+		`;
+	}
+
+	update(changedProperties) {
+		super.update(changedProperties);
+
+		if (changedProperties.get('autoShow')) {
```

Comment:
This will only run if `autoShow` changed to `true` (not from true => false) -- is that the intent?

Reply:
Same situation as above where I don't know what the original intent was but for this case I think it is correct as is. The other case where `_autoDetectBoundaries` is run also is only run when `autoShow` is true

Reply:
So did some investigation and `get` actually returns the old value not the new, so this code runs when `autoShow` goes from true to false, which I don't think was the intention. It actually throws in the case where it goes from true to false, so I'll do some fixing there.